File: Avion.cpp
// Archivo Avion.cpp

#include "Avion.h"


// Implementación de los métodos de la clase Avion
Avion::Avion(string tipo, string marca, string modelo, int Num_Asientos, int velo_max, int autonomia, int agno, string estado, int altura_max, int no_motores, string cat)
    : Vehiculos_Aereos(tipo, marca, modelo, Num_Asientos, velo_max, autonomia, agno, estado), Altura_Max(altura_max), No_Motores(no_motores), categoria(cat) {
    }

int Avion::get_Altura_Max() const {
    return Altura_Max;
}

void Avion::set_Altura_Max(int New_Alt) {
    Altura_Max = New_Alt;
}

int Avion::get_No_Motores() const {
    return No_Motores;
}

void Avion::set_No_Motores(int New_Mot) {
    No_Motores = New_Mot;
}

string Avion::get_categoria() {
    return categoria;
}

void Avion::set_categoria(bool New_Cat) {
    categoria = New_Cat;
}

void Avion::Show_Info_A(){
	Show_Info();
	cout << "Altura Maxima: " << Altura_Max << endl;
    cout << "Numero de Motores: " << No_Motores << endl;
    cout << "Categoria: " << categoria << endl;
    cout << endl;
}

File: proyectoAeropuertoPython/avion.py
from VehiculosAereos import VehiculosAereos  
class Avion(VehiculosAereos):
    def __init__(self, tipo, marca, modelo, num_asientos, velo_max, autonomia, agno, estado, altura_max, no_motores, cat):
        super().__init__(tipo, marca, modelo, num_asientos, velo_max, autonomia, agno, estado)
        self.Altura_Max = altura_max
        self.No_Motores = no_motores
        self.categoria = cat

    def get_Altura_Max(self):
        return self.Altura_Max

    def set_Altura_Max(self, new_alt):
        self.Altura_Max = new_alt

    def get_No_Motores(self):
        return self.No_Motores

    def set_No_Motores(self, new_mot):
        self.No_Motores = new_mot

    def get_categoria(self):
        return self.categoria

    def set_categoria(self, new_cat):
        self.categoria = new_cat

    def Show_Info_A(self):
        self.Show_Info()
        print("Altura Maxima:", self.Altura_Max)
        print("Numero de Motores:", self.No_Motores)
        print("Categoria:", self.categoria)